account-kit/react/src/hooks/useUiConfig.ts

```diff
@@ -1,29 +1,91 @@
 "use client";
 
+import { useEffect, useMemo } from "react";
+import { create } from "zustand";
 import { useAlchemyAccountContext } from "../context.js";
 import { MissingUiConfigError } from "../errors.js";
-import type { AlchemyAccountsUIConfig } from "../types.js";
+import type {
+  AlchemyAccountsUIConfig,
+  AuthIllustrationStyle,
+} from "../types.js";
 
+type AlchemyAccountsUIConfigWithDefaults = Omit<
+  Required<AlchemyAccountsUIConfig>,
+  "auth"
+> & {
+  auth: Exclude<Required<AlchemyAccountsUIConfig["auth"]>, undefined>;
+};
+
+type UiConfigStore = AlchemyAccountsUIConfig & {
+  updateConfig: (partial: AlchemyAccountsUIConfig) => void;
+};
+
+export const DEFAULT_UI_CONFIG: AlchemyAccountsUIConfigWithDefaults = {
+  illustrationStyle: "flat" as AuthIllustrationStyle,
+  auth: {
+    addPasskeyOnSignup: false,
+    header: null,
+    hideError: false,
+    sections: [[{ type: "email" }], [{ type: "passkey" }]],
+    showSignInText: true,
+    onAuthSuccess: () => {},
+  },
+  modalBaseClassName: "",
+};
+
+const internal_useUiConfig = create<UiConfigStore>((set) => ({
```

Comment:
this might be overkill, but I don't wanna deal with partial state updates that I'd have to deal with if I used `useState`